### ISSUE TITLE
podman_network ignore dns when using quadlet

### ISSUE THREAD

**Issue by giovannibaratta** [containers/ansible-podman-collections#792]
**Is this a BUG REPORT or FEATURE REQUEST?**

/kind bug

**Description**

The attribute DNS is not included in the generated quadlet for the module podman_network

1. Define podman_network
    ```yaml
    ---
    - hosts: all
    
      tasks:
      - name: Create Coder network Quadlet
        containers.podman.podman_network:
          name: test-no-quad2
          state: quadlet
          dns: 172.16.255.254
          driver: bridge
    ```

2. ansible-bug ansible-playbook bug-with-quadlet.yaml -i inventory.yaml -v

    ```bash
    Using /etc/ansible/ansible.cfg as config file
    
    PLAY [all] ************************************************************************************************************************************************************
    
    TASK [Gathering Facts] ************************************************************************************************************************************************
    ok: [172.16.255.254]
    
    TASK [Create Coder network Quadlet] ***********************************************************************************************************************************
    changed: [172.16.255.254] => {"actions": [], "changed": true, "network": {}}
    
    PLAY RECAP ************************************************************************************************************************************************************
    172.16.255.254             : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
    ```
3. Start the service and inspect the result

    ```bash
    ansible@homelab-01:~$ cat $HOME/.config/containers/systemd/test-no-quad2.network
    [Network]
    NetworkName=test-no-quad2
    Driver=bridge
    
    ansible@homelab-01:~$ systemctl daemon-reload --user
    ansible@homelab-01:~$ systemctl start --user test-no-quad2-network.service
    ansible@homelab-01:~$ podman network inspect test-no-quad2 | jq '.[0].network_dns_servers'
    null
    ```

**Describe the results you received:**

```bash
podman network inspect test-no-quad2 | jq '.[0].network_dns_servers'
null
```

**Describe the results you expected:**

```bash
podman network inspect test-no-quad | jq '.[0].network_dns_servers'
[
  "172.16.255.254"
]
```

For example, the configuration above can be obtained with 

```yaml
---
- hosts: all

  tasks:
  - name: Create Coder network Quadlet
    containers.podman.podman_network:
      name: test-no-quad
      state: present
      dns: 172.16.255.254
      driver: bridge
```

**Version of the `containers.podman` collection:**

```
ansible-galaxy collection list | grep containers.podman
containers.podman                        1.15.3
containers.podman                        1.13.0
```

**Output of `ansible --version`:**

```
ansible [core 2.16.7]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/gbaratta/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  ansible collection location = /home/gbaratta/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] (/usr/bin/python3)
  jinja version = 3.0.3
  libyaml = True
```

**Output of `podman version`:**

```
ansible@homelab-01:~$ podman version
Client:       Podman Engine
Version:      5.1.1
API Version:  5.1.1
Go Version:   go1.22.4
Built:        Thu Jan  1 00:00:00 1970
OS/Arch:      linux/amd64
```

<details>
<summary>Output of podman info --debug:</summary>

``` yaml
host:
  arch: amd64
  buildahVersion: 1.36.0
  cgroupControllers:
  - cpu
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon_100:2.1.12-1_amd64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.12, commit: e21e7c85b7637e622f21c57675bf1154fc8b1866'
  cpuUtilization:
    idlePercent: 99.47
    systemPercent: 0.06
    userPercent: 0.47
  cpus: 16
  databaseBackend: sqlite
  distribution:
    codename: noble
    distribution: ubuntu
    version: "24.04"
  eventLogger: journald
  freeLocks: 2048
  hostname: homelab-01
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1001
      size: 1
    - container_id: 1
      host_id: 165536
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1001
      size: 1
    - container_id: 1
      host_id: 165536
      size: 65536
  kernel: 6.8.0-35-generic
  linkmode: dynamic
  logDriver: journald
  memFree: 49437564928
  memTotal: 66591793152
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns_1.4.0-5_amd64
      path: /usr/lib/podman/aardvark-dns
      version: aardvark-dns 1.4.0
    package: Unknown
    path: /usr/libexec/podman/netavark
    version: netavark 1.11.0
  ociRuntime:
    name: crun
    package: crun_100:1.15-1_amd64
    path: /usr/bin/crun
    version: |-
      crun version 1.15
      commit: e6eacaf4034e84185fd8780ac9262bbf57082278
      rundir: /run/user/1001/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt_100:0.0+20240607.8a83b530-1_amd64
    version: |
      pasta 0.0+20240607.8a83b530
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: false
    path: /run/user/1001/podman/podman.sock
  rootlessNetworkCmd: pasta
  security:
    apparmorEnabled: false
    capabilities: CAP_AUDIT_WRITE,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_MKNOD,CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns_100:1.3.1-1_amd64
    version: |-
      slirp4netns version 1.3.1
      commit: unknown
      libslirp: 4.8.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.5
  swapFree: 8589930496
  swapTotal: 8589930496
  uptime: 84h 35m 51.00s (Approximately 3.50 days)
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /home/ansible/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/ansible/.local/share/containers/storage
  graphRootAllocated: 52521566208
  graphRootUsed: 27438362624
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Supports shifting: "false"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 0
  runRoot: /run/user/1001/containers
  transientStore: false
  volumePath: /home/ansible/.local/share/containers/storage/volumes
version:
  APIVersion: 5.1.1
  Built: 0
  BuiltTime: Thu Jan  1 00:00:00 1970
  GitCommit: ""
  GoVersion: go1.22.4
  Os: linux
  OsArch: linux/amd64
  Version: 5.1.1
```
</details>

**Comment by giovannibaratta** [comment]
From my understanding, the problem should be in the `param_map` of the [NetworkQuadlet](https://github.com/containers/ansible-podman-collections/blob/d113847336e3cea9b2bef5a204f6f40676476afe/plugins/module_utils/podman/quadlet.py#L389C9-L389C22). The key for the dns should be `dns` instead of `DNS`.

**Comment by sshnaidm** [comment]
> From my understanding, the problem should be in the `param_map` of the [NetworkQuadlet](https://github.com/containers/ansible-podman-collections/blob/d113847336e3cea9b2bef5a204f6f40676476afe/plugins/module_utils/podman/quadlet.py#L389C9-L389C22). The key for the dns should be `dns` instead of `DNS`.

@giovannibaratta would you like to prepare a patch?

**Comment by giovannibaratta** [comment]
Sure

**Comment by giovannibaratta** [comment]
Fixed in #793
